training/src/main/scala/com/airbnb/aerosolve/training/pipeline/GenericPipeline.scala

```diff
@@ -509,14 +529,14 @@ object GenericPipeline {
       .map(x => hiveTrainingToExample(x, schema, isMulticlass))
   }
 
-  def isTraining(examples : Example) : Boolean = {
+  def isTraining(frac: Double = DEFAULT_TRAINING_FRACTION)(examples : Example) : Boolean = {
     // Take the hash code mod 255 and keep the first 16 as holdout.
-    (examples.toString.hashCode & 0xFF) > 16
+    (examples.toString.hashCode & 0xFF) > 255 - Math.floor(255 * frac)
   }
```

Comment:
Interesting idea! Is there a reason why we don't use just `examples.hashCode` directly?